src/queries/incremental-compilation-in-detail.md

```diff
@@ -401,6 +401,97 @@ deal with all of the above but so far that seemed like more trouble than it
 would save.
 
 
+
+## Query Modifiers
+
+The query system allows for applying [modifiers][mod] to queries. These
+modifiers affect certain aspects of how the system treats the query with
+respect to incremental compilation:
+
+ - `eval_always` - A query with the `eval_always` attribute is re-executed
+   unconditionally during incremental compilation. I.e. the system will not
+   even try to mark the query's dep-node as green. This attribute has two use
+   cases:
+
+    - Queries that are **inputs** to the system must be marked as `eval_always`
+      since we have no other way of checking whether their value has changed.
```

Comment:
I think it's more accurate to phrase this as `eval_always` queries can read inputs (from files, global state, etc). They can also produce side effects like changing global state and writing to files.